inc/based_config.php

```diff
@@ -44,118 +44,81 @@
-if (!defined("GLPI_PLUGIN_DOC_DIR")) {
-   define("GLPI_PLUGIN_DOC_DIR", GLPI_VAR_DIR . "/_plugins");
-}
-// Path for cache storage
-if (!defined("GLPI_LOCK_DIR")) {
-   define("GLPI_LOCK_DIR", GLPI_VAR_DIR . "/_lock");
-}
-
-// Path for log storage
-if (!defined("GLPI_LOG_DIR")) {
-   define("GLPI_LOG_DIR", GLPI_VAR_DIR . "/_log");
-}
-
-// Path for graph storage
-if (!defined("GLPI_GRAPH_DIR")) {
-   define("GLPI_GRAPH_DIR", GLPI_VAR_DIR . "/_graphs");
-}
-
-// Path for picture storage
-if (!defined("GLPI_PICTURE_DIR")) {
-   define("GLPI_PICTURE_DIR", GLPI_VAR_DIR . "/_pictures");
-}
-
-// Path for temp storage
-if (!defined("GLPI_TMP_DIR")) {
-   define("GLPI_TMP_DIR", GLPI_VAR_DIR . "/_tmp");
-}
-
-// Path for cache
-if (!defined("GLPI_CACHE_DIR")) {
-   define("GLPI_CACHE_DIR", GLPI_VAR_DIR . "/_cache");
-}
-
-// Path for rss storage
-if (!defined("GLPI_RSS_DIR")) {
-   define("GLPI_RSS_DIR", GLPI_VAR_DIR . "/_rss");
-}
-
-// Path for upload storage
-if (!defined("GLPI_UPLOAD_DIR")) {
-   define("GLPI_UPLOAD_DIR", GLPI_VAR_DIR . "/_uploads");
-}
+(function() {
+   // Define GLPI_* constants that can be customized by admin.
+   //
+   // Use a self-invoking anonymous function to:
+   // - prevent any global variables/functions definition from `local_define.php` and `downstream.php` files;
+   // - prevent any global variables definition from current function logic.
+
+   $constants = [
+      'GLPI_CONFIG_DIR'     => GLPI_ROOT . '/config', // Path for configuration files (db, security key, ...)
+      'GLPI_VAR_DIR'        => GLPI_ROOT . '/files',  // Path for all files
+      'GLPI_DOC_DIR'        => '{GLPI_VAR_DIR}', // Path for documents storage
+      'GLPI_CACHE_DIR'      => '{GLPI_VAR_DIR}/_cache', // Path for cache
+      'GLPI_CRON_DIR'       => '{GLPI_VAR_DIR}/_cron', // Path for cron storage
+      'GLPI_DUMP_DIR'       => '{GLPI_VAR_DIR}/_dumps', // Path for backup dump
+      'GLPI_GRAPH_DIR'      => '{GLPI_VAR_DIR}/_graphs', // Path for graph storage
+      'GLPI_LOCAL_I18N_DIR' => '{GLPI_VAR_DIR}/_locales', // Path for local i18n files
+      'GLPI_LOCK_DIR'       => '{GLPI_VAR_DIR}/_lock', // Path for lock files storage (used by cron)
+      'GLPI_LOG_DIR'        => '{GLPI_VAR_DIR}/_log', // Path for log storage
+      'GLPI_PICTURE_DIR'    => '{GLPI_VAR_DIR}/_pictures', // Path for picture storage
+      'GLPI_PLUGIN_DOC_DIR' => '{GLPI_VAR_DIR}/_plugins', // Path for plugins documents storage
+      'GLPI_RSS_DIR'        => '{GLPI_VAR_DIR}/_rss', // Path for rss storage
+      'GLPI_SESSION_DIR'    => '{GLPI_VAR_DIR}/_sessions', // Path for sessions storage
+      'GLPI_TMP_DIR'        => '{GLPI_VAR_DIR}/_tmp', // Path for temp storage
+      'GLPI_UPLOAD_DIR'     => '{GLPI_VAR_DIR}/_uploads', // Path for upload storage
+
+      'GLPI_INSTALL_MODE'   => is_dir(GLPI_ROOT . '/.git') ? 'GIT' : 'TARBALL', // Install mode for telemetry
+
+      // TODO Check if following constants can be removed
```

Comment:
As discussed, font was used in old versions of `inc/stat.class.php`, can be removed. I did not find any ocurrence of `GLPI_SCRIPT_DIR` in 9.1/bf and 0.90/bf branches... I guess we can remove it as well.